src/graphics/shapes/ellipse.rs

```diff
@@ -6,37 +6,47 @@ use std::rc::Rc;
 use web_sys::WebGl2RenderingContext;
 
 pub struct Ellipse {
+    pub x: i32,
+    pub y: i32,
     pub width: f32,
     pub height: f32,
 
     geom: Rc<RefCell<Geom>>,
 }
 pub struct Circle {
+    pub x: i32,
+    pub y: i32,
     pub radius: f32,
 
     geom: Rc<RefCell<Geom>>,
 }
 
 impl Ellipse {
-    pub fn new(width: f32, height: f32, color: &Vec<f32>) -> Self {
+    pub fn new(x: i32, y: i32, width: f32, height: f32, color: &Vec<f32>) -> Self {
         let vertex_count = 200;
         let vertices = calc_n_vertices(width, height, vertex_count);
         let color_data = color_n_vertices(color, vertex_count);
 
         let geom = Rc::new(RefCell::new(Geom {
             vertices,
             color: color_data,
-            u_mat: Matrix::new(),
+            u_mat: Matrix::translation(x as f32, y as f32),
```

Comment:
I think translation is different from relative distance to (0,0) point in viewport

Reply:
Now thinking about this I think bounding boxes are necessary for this as well